main.tf

```diff
@@ -59,6 +67,63 @@ locals {
   )
 }
 
+# ---------------------------------------------------------------------------------------------------------------------
+# ¦ LAMBDA SQS TRIGGER (OPTIONAL)
+# ---------------------------------------------------------------------------------------------------------------------
+resource "aws_sqs_queue" "lambda_trigger" {
+  count = local.trigger_sqs == true ? 1 : 0
+
+  name                       = local.trigger_sqs_name
+  visibility_timeout_seconds = var.timeout
+  tags                       = var.resource_tags
+}
+
+resource "aws_sqs_queue_policy" "lambda_trigger" {
+  count = local.trigger_sqs == true ? 1 : 0
+
+  queue_url = aws_sqs_queue.lambda_trigger[0].id
+  policy    = data.aws_iam_policy_document.lambda_trigger[0].json
+}
+
+data "aws_iam_policy_document" "lambda_trigger" {
+  count = local.trigger_sqs == true ? 1 : 0
+
+  statement {
+    sid     = "EnableIamUserPermissions"
+    actions = ["sqs:*"]
+    effect  = "Allow"
+    principals {
+      type        = "AWS"
+      identifiers = [format("arn:aws:iam::%s:root", data.aws_caller_identity.current.account_id)]
+    }
+    resources = ["*"]
+  }
+  statement {
+    sid = "AllowOutboundSns"
+    actions = [
+      "sqs:SendMessage"
+    ]
+    principals {
+      type        = "Service"
+      identifiers = ["sns.amazonaws.com"]
+    }
+    resources = ["*"]
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+      values = var.triggering_sns_arns      
+    }
+  }
+}
+
+resource "aws_lambda_event_source_mapping" "security_findings_enriched_sqs_trigger" {
+  count = local.trigger_sqs == true ? 1 : 0
+
```

Comment:
should not be named "security_findings_enriched_sqs_trigger" as this will be a general feature